### DISCUSSION TITLE
error patching application with operation state: Request entity too large: limit is 3145728

### DISCUSSION THREAD

**Discussion by jdomag** [argoproj/argo-cd#16462] [General]
Hi, 
I have an Application that renders helm chart. The helm chart builds around 150 configmaps with 500 keys in each of them. Individual configmap is ~30Kb in size, all the configmaps from helm chart are around ~2Mb. When I try to Sync the App it doesn't sync with the following error from application controller:
`{"appNamespace":"argocd","application":"my-properties","level":"warning","msg":"error patching application with operation state: Request entity too large: limit is 3145728","project":"staging"}`
I'm trying to understand what the error in application controller means. 
What does the app controller tries to do? Is it about the limit of App object or individual confgimap?

Interesting fact is that same helm chart on different cluster sync without an issue.

**Comment by mswenson-redhat** [comment]
@jdomag did you ever find a solution to this?

**Comment by rumstead** [comment]
That is an etcd limit. What in your chart is 3mb?

**Reply by mswenson-redhat** [reply]
@rumstead My chart is applying 17,000 manifests. I think this limitation is actually how large the application CRD can be since it is trying to write the status of every manifest into the CRD. That may be a limitation based on etcd to your point. I guess the question here becomes how many manifests can an Argo Application realistically support and/or is there a way we can condense the Argo status similarly to how Helm does it.

**Reply by rumstead** [reply]
Wow, that is a lot of manifests (your poor, poor Argo UI if you try to load that application 😆 )! I see that makes sense. The application quite literally itemizes each resource and in some cases the health status. Maybe it would be worthwhile to create a feature request? I don't think there is a way around it currently.
